hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -587,7 +587,8 @@ private List<StateChanges.Builder> onMigrate(
         migrationStateChanges.addAll(migrationChanges);
         kvStateChangeListener.reset();
         boundaryStateChangeListener.reset();
-        if (isUpgrade && !trigger.equals(RECONNECT)) {
+        // For specifically a non-genesis upgrade, set in state that post-upgrade work is pending
+        if (isUpgrade && trigger != RECONNECT && trigger != GENESIS) {
```

Comment:
Was the non-genesis case actually needed or was it just an oversight and this new code is "tighter"?